src/devices/bus/a2bus/a2superdrive.cpp

```diff
@@ -0,0 +1,361 @@
+	map(0x8000, 0xffff).rom().region(SUPERDRIVE_ROM_REGION, 0x0000);
+}
+
+
+void a2bus_superdrive_device::device_add_mconfig(machine_config &config)
+{
+
+	M65C02(config, m_65c02,  A2BUS_7M_CLOCK / 3.5);  /* ~ 2.046 MHz */
+	m_65c02->set_addrmap(AS_PROGRAM, &a2bus_superdrive_device::m65c02_mem);
+
+	SWIM1(config, m_fdc, C16M);
+
+	for (auto &floppy: m_floppy)
+		applefdintf_device::add_35_hd(config, floppy);
+
+
+	m_fdc->devsel_cb().set(FUNC(a2bus_superdrive_device::devsel_w));
+	m_fdc->hdsel_cb().set(FUNC(a2bus_superdrive_device::hdsel_w));
+	m_fdc->phases_cb().set(FUNC(a2bus_superdrive_device::phases_w));
+	m_fdc->sel35_cb().set(FUNC(a2bus_superdrive_device::sel35_w));
+}
+
+
+
+a2bus_superdrive_device::a2bus_superdrive_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, type, tag, owner, clock),
+	device_a2bus_card_interface(mconfig, *this),
+	m_65c02(*this, "superdrive_65c02"),
+	m_rom(*this, SUPERDRIVE_ROM_REGION),
+	m_ram(*this, SUPERDRIVE_RAM_TAG),
+	m_fdc(*this, "fdc"),
+	m_floppy(*this, "%u", 0U),
+	m_cur_floppy(nullptr),
+	m_bank_select(0),
+	m_side(0)
+{ }
+
+a2bus_superdrive_device::a2bus_superdrive_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock):
+	a2bus_superdrive_device(mconfig, A2BUS_SUPERDRIVE, tag, owner, clock)
+{ }
+
+void a2bus_superdrive_device::device_start()
+{
+	save_item(NAME(m_bank_select));
+	save_item(NAME(m_side));
+}
+
+
+void a2bus_superdrive_device::device_reset()
+{
+	m_bank_select = 0;
+	m_side = 0;
+	m_65c02->reset();
+}
+
+
+// overrides of standard a2bus slot functions
+uint8_t a2bus_superdrive_device::read_c0nx(uint8_t offset)
+{
+	if (machine().side_effects_disabled()) return 0;
+
+	m_bank_select = offset & 0x0f;
+	return 0;
+}
+
+void a2bus_superdrive_device::write_c0nx(uint8_t offset, uint8_t data)
+{
+	if (machine().side_effects_disabled()) return;
+
+	m_bank_select = offset & 0x0f;
+}
+
+uint8_t a2bus_superdrive_device::read_cnxx(uint8_t offset)
+{
+	return m_ram[0x7b00 + offset];
+}
+
+void a2bus_superdrive_device::write_cnxx(uint8_t offset, uint8_t data)
+{
+	//m_ram[0x7b00 + offset] = data;
+}
+
+/*
+ * $c800 - $cbff is uc RAM, controlled via c0nx
+ * $cc00 - $cfff is uc RAM, hardcoded to $7c00 - $7fff
+ */
+uint8_t a2bus_superdrive_device::read_c800(uint16_t offset)
+{
+
+	unsigned address;
+
+	if (offset < 0x400)
+		address = (m_bank_select << 10) + offset;
+	else
+		address = 0x7c00 + offset - 0x400;
+
+	return m_ram[address];
+}
+
+void a2bus_superdrive_device::write_c800(uint16_t offset, uint8_t data)
+{
+
+	unsigned address;
+
+	if (offset < 0x400)
+		address = (m_bank_select << 10) + offset;
+	else
+		address = 0x7c00 + offset - 0x400;
+
+	m_ram[address] = data;
+}
+
+
+/* uc 65c02 i/o at $0a00 */
+void a2bus_superdrive_device::m65c02_w(offs_t offset, uint8_t value)
+{
+	if (machine().side_effects_disabled()) return;
```

Comment:
This is another case where suppressing the effects of the write is going to confuse people.